src/main/java/com/marklogic/mule/extension/WriteOperations.java

```diff
@@ -16,44 +16,63 @@
 package com.marklogic.mule.extension;
 
 import com.marklogic.client.DatabaseClient;
-import com.marklogic.client.document.GenericDocumentManager;
 import com.marklogic.client.io.DocumentMetadataHandle;
 import com.marklogic.client.io.Format;
 import com.marklogic.client.io.InputStreamHandle;
 
 import java.io.InputStream;
 import java.util.UUID;
 
+import static com.marklogic.client.io.Format.XML;
+
 public class WriteOperations {
     void writeDocuments(DatabaseClient databaseClient, InputStream content,
                         Format format, String permissions, int quality,
-                        String collections, String uriPrefix, String uriSuffix, boolean generateUUID) {
-
-        GenericDocumentManager documentManager = databaseClient.newDocumentManager();
+                        String collections, String uriPrefix, String uriSuffix, boolean generateUUID, String temporalCollection,
+                        String restTransform, String restTransformParameters, String restTransformParametersDelimiter) {
 
         DocumentMetadataHandle documentMetadataHandle = new DocumentMetadataHandle().withQuality(quality);
 
-        if (collections != null && !collections.isEmpty()) {
+        if (Utilities.hasText(collections)) {
             String[] collectionArray = collections.split(",");
             documentMetadataHandle.withCollections(collectionArray);
         }
 
-        if (permissions != null && !permissions.isEmpty()) {
+        if (Utilities.hasText(permissions)) {
             documentMetadataHandle.getPermissions().addFromDelimitedString(permissions);
         }
         StringBuilder uri = new StringBuilder();
-        if (uriPrefix != null && !uriPrefix.isEmpty()) {
+        if (Utilities.hasText(uriPrefix)) {
             uri.append(uriPrefix);
         }
         if (generateUUID) {
             uri.append(UUID.randomUUID());
         }
-        if (uriSuffix != null && !uriSuffix.isEmpty()) {
+        if (Utilities.hasText(uriSuffix)) {
             uri.append(uriSuffix);
         }
-        documentManager
-            .write(uri.toString(),
-                documentMetadataHandle,
-                new InputStreamHandle(content).withFormat(format));
+        if(Utilities.hasText(temporalCollection)){
```

Comment:
Small thing - the code for creating the content handle and the server transform is in 3 places. To make this read better and avoid the duplication, use a couple variables:

```
InputStreamHandle contentHandle = new InputStreamHandle(content).withFormat(format);
ServerTransform transform = Utilities.findServerTransform(restTransform, ...)
```